raster-foundry/raster-foundry
# Fix routing problem for scene detail tiles

## Overview

This PR dramatically DRYs out middleware application in `backsplash-server`, eliminates some unnecessary middlewares, and kills a layer of matching in server routes.

### Checklist


### Demo

![scene-tiles](https://user-images.githubusercontent.com/5702984/66237137-f4afa200-e6b9-11e9-8b36-e4fd84d9c60f.gif)

### Notes

I wanted it also to use `AutoSlash` from `http4s.server.middleware` but I couldn't figure out how to make that work. Given how much else I was able to light on :fire: I'm not worrying too much about it.

The dramatic changes to middleware / http4s upgrade were to get the API to look as much like http4s documentation as possible and to decrease the surface area of "things that could be different between rendering scenes and projects". View changes without whitespace changes to get a sense of how much was trimmed.

## Testing Instructions

- `;reload;backsplash-server/assembly` (or just `backsplash-server/assembly` if you don't already have an sbt shell open)
- click on an ingested scene from browsing a project
- you should get tiles

Closes azavea/raster-foundry-platform#857
